src/commcare_cloud/commands/deploy/formplayer.py

```diff
@@ -102,7 +102,7 @@ def get_deploy_diff(environment, repo):
 
 
 def create_release_tag(environment, repo, diff):
-    if not github_auth_provided():
+    if not github_auth_provided() or not environment.fab_settings_config.tag_deploy_commits:
```

Comment:
Think this should be a hard requirement?  This is consistent with existing behavior, so a change would be out-of-scope for this PR, but I could see us refusing to deploy a `tag_deploy_commits` env unless you provide auth (to prevent silent deploys that don't show up properly).